collectors/perf.go

```diff
@@ -406,233 +180,72 @@ func (c *PerfCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.DiskBytesPerSec,
-			prometheus.GaugeValue,
-			float64(volume.DiskBytesPerSec),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.DiskReadBytesPerSec,
+			c.ReadBytesPerSec,
 			prometheus.GaugeValue,
 			float64(volume.DiskReadBytesPerSec),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.DiskReadsPerSec,
+			c.ReadsPerSec,
 			prometheus.GaugeValue,
 			float64(volume.DiskReadsPerSec),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.DiskTransfersPerSec,
-			prometheus.GaugeValue,
-			float64(volume.DiskTransfersPerSec),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.DiskWriteBytesPerSec,
+			c.WriteBytesPerSec,
 			prometheus.GaugeValue,
 			float64(volume.DiskWriteBytesPerSec),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.DiskWritesPerSec,
+			c.WritesPerSec,
 			prometheus.GaugeValue,
 			float64(volume.DiskWritesPerSec),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.FreeMegabytes,
-			prometheus.GaugeValue,
-			float64(volume.FreeMegabytes),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskReadTime,
-			prometheus.GaugeValue,
-			float64(volume.PercentDiskReadTime),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskReadTime_Base,
-			prometheus.GaugeValue,
-			float64(volume.PercentDiskReadTime_Base),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskTime,
-			prometheus.GaugeValue,
-			float64(volume.PercentDiskTime),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskTime_Base,
-			prometheus.GaugeValue,
-			float64(volume.PercentDiskTime_Base),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskWriteTime,
-			prometheus.GaugeValue,
-			float64(volume.PercentDiskWriteTime),
-			volume.Name,
-		)
-
-		ch <- prometheus.MustNewConstMetric(
-			c.PercentDiskWriteTime_Base,
+			c.FreeBytes,
 			prometheus.GaugeValue,
-			float64(volume.PercentDiskWriteTime_Base),
+			float64(volume.FreeMegabytes)*1024*1024,
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.PercentFreeSpace,
+			c.ReadTime,
 			prometheus.GaugeValue,
-			float64(volume.PercentFreeSpace),
+			float64(volume.PercentDiskReadTime)/float64(volume.PercentDiskReadTime_Base),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.PercentFreeSpace_Base,
+			c.WriteTime,
 			prometheus.GaugeValue,
-			float64(volume.PercentFreeSpace_Base),
+			float64(volume.PercentDiskWriteTime)/float64(volume.PercentDiskWriteTime_Base),
 			volume.Name,
 		)
 
 		ch <- prometheus.MustNewConstMetric(
-			c.PercentIdleTime,
+			c.FreeSpace,
 			prometheus.GaugeValue,
-			float64(volume.PercentIdleTime),
+			float64(volume.PercentFreeSpace)/float64(volume.PercentFreeSpace_Base),
```

Comment:
Hm, how do you mean? The values I'm getting from this division seem correct? Or did you mean to split it into total space and available space?